File: solution/1422. Maximum Score After Splitting a String/1422_02.cpp
class Solution {
public:
    int maxScore(string s) {
        int res = 0, zeroL = 0, oneR = count(s.begin(), s.end(), '1');
        for (int i = 0; i < s.size() - 1; ++i) {
            if (s[i] == '0') ++zeroL;
            else --oneR;
            res = max(res, zeroL + oneR);
        }
        return res;
    }
};

File: solution/1422. Maximum Score After Splitting a String/1422_02.py
class Solution:
    def maxScore(self, s: str) -> int:
        res = zero_left = 0
        one_right = s.count('1')
        
        for i in range(len(s) - 1):
            if s[i] == '0':
                zero_left += 1
            else:
                one_right -= 1
            res = max(res, zero_left + one_right)
        
        return res
    
